flight/Libraries/math/coordinate_conversions.c

```diff
@@ -73,11 +73,35 @@ void Quaternion2RPY(const float q[4], float rpy[3])
 	R23 = 2.0f * (q[2] * q[3] + q[0] * q[1]);
 	R33 = q0s - q1s - q2s + q3s;
 
+	/* If we're nearly nose-up or nose-down, consider yaw to be 0, encode
+	 * information in roll to deal with loss of freedom
+	 */
```

Comment:
Wouldn't it be of interest to implement a transition zone, where it fades between presumed and zero yaw based on angle? I mean, it's all an edge case to begin with, but there might be a large yaw error transitioning out of that extreme pitched attitude. If that yaw value is consumed by a controller, there might be an issue?

Reply:
Doesn't look like it.  Play with the function.  Arguably we should have a higher threshold than .985, but I can't make this be off by more than .8 degrees.  The thing is, once you're >80 degrees nose up, roll almost entirely = -yaw and vice versa

Reply:
Note that this is used for two main things-- A) converting attitude to R/P/Y which is not used by controllers anymore but is used by the PFD in GCS and the OSD, and B) converting an attitude error into R/P/Y errors.

In the case of A, .. there's a singularity there no matter what-- no matter what the solution is numerically unstable.  Tiny movements in orientation can create huge changes in roll & yaw (which are really the same parameter).

In the case of B, what this really means is, "if we can't tell the difference between roll & yaw, start off making the change entirely with roll".  And roll was chosen here because we might not control vehicle yaw.  Eliminating roll when we have no control of yaw would be weird.